### ISSUE TITLE
Problems getting CTC to work

### ISSUE THREAD

**Issue by queader** [Microsoft/CNTK#2335]
Hello, I made a simple test network that uses CTC (forward_backward) based on the tutorial and I'm getting strange result that I don't understand. My model takes a sequence of one hot encoded characters and as labels I give it the same sequence, so I think the result should also be the same sequence. However, I can't get it to converge and output anything other that blank labels only. Below I include example that shows this behaviour:

```
import os
import cntk as C
import numpy as np
import random

alphabet = "-0123456789abcdefghijklmnopqrstuvwxyz"

feature_dimension = label_dimension = len(alphabet)
feature = C.sequence.input((feature_dimension))
label = C.sequence.input((label_dimension))

with C.default_options(activation=C.sigmoid):
    z = C.layers.Sequential([
        C.layers.For(range(3), lambda: C.layers.Recurrence(C.layers.LSTM(1024))),
        C.layers.Dense(label_dimension)
    ])(feature)

criteria = C.forward_backward(C.labels_to_graph(label), z, blankTokenId=0)
err = C.edit_distance_error(z, label, squashInputs=True, tokensToIgnore=[0])

lr = C.learning_rate_schedule(0.01, C.UnitType.sample)
learner = C.adadelta(z.parameters, lr)
progress_printer = C.logging.progress_print.ProgressPrinter(50, tag="Training")
trainer = C.Trainer(z, (criteria, err), learner, progress_writers=[progress_printer])

one_hot_arr = np.eye(label_dimension)

def encode_text(text):
    return [one_hot_arr[alphabet.index(l)] for l in text]

def decode_text(one_hot):
    return ''.join([alphabet[np.argmax(i)] for i in one_hot])

def next_minibatch_el():
    str = ''.join(random.choice(alphabet) * 4 for _ in range(8))
    encoded = encode_text(str)
    return encoded, encoded, str

def next_minibatch(batch_size):
    batchX = []
    batchY = []
    raw_labels = []

    for i in range(batch_size):
        img, label, label_raw = next_minibatch_el()
        batchX.append(img)
        batchY.append(label)
        raw_labels.append(label_raw)

    return batchX, batchY, raw_labels

def val_preview():
    xL, yL, gtL = next_minibatch(64)

    for x1, gt1 in list(zip(xL, gtL))[:15]:
        result = z.eval({feature: [x1]})[0]
        pred = decode_text(result)
        print('network out: %-32s => ground truth: %-32s' % (pred, gt1))

i = 0
while True:
    x1, y1, gt1 = next_minibatch(64)
    trainer.train_minibatch({feature: x1, label: y1})

    if i % 50 == 0:
        val_preview()

    i += 1
```

This is a sample from the output it produces, notice how the loss doesn't improve at all at this point:

```
Minibatch[1001-1050]: loss = 2.656250 * 102400, metric = 100.00% * 102400;
network out: -------------------------------- => ground truth: iiiiiiiillllgggguuuuaaaappppjjjj
network out: -------------------------------- => ground truth: bbbbaaaahhhhsssskkkkjjjjwwwwiiii
network out: -------------------------------- => ground truth: iiiiyyyywwww0000xxxxddddjjjjwwww
network out: -------------------------------- => ground truth: ffffjjjjkkkkxxxx66667777iiiitttt
network out: -------------------------------- => ground truth: 44446666kkkkuuuu4444ttttzzzzhhhh
network out: -------------------------------- => ground truth: ccccrrrr----ddddllll6666yyyy0000
network out: -------------------------------- => ground truth: eeeeaaaa222255557777ppppeeeehhhh
network out: -------------------------------- => ground truth: 666666665555aaaauuuuwwwwrrrrqqqq
network out: -------------------------------- => ground truth: 5555nnnn2222wwww11115555eeeeyyyy
network out: -------------------------------- => ground truth: ssss4444llll55558888bbbbaaaaxxxx
network out: -------------------------------- => ground truth: 3333hhhhoooo5555iiii77770000gggg
network out: -------------------------------- => ground truth: 3333iiiibbbbzzzzjjjj7777gggg5555
network out: -------------------------------- => ground truth: rrrr7777xxxxggggxxxxooooiiiiqqqq
network out: -------------------------------- => ground truth: 9999mmmmppppllllkkkkiiiibbbb2222
network out: -------------------------------- => ground truth: ddddsssskkkkccccoooo2222hhhheeee
 Minibatch[1051-1100]: loss = 2.656250 * 102400, metric = 100.00% * 102400;
network out: -------------------------------- => ground truth: llllbbbbiiiidddd777799994444ssss
network out: -------------------------------- => ground truth: rrrrnnnnpppp55554444pppp0000gggg
network out: -------------------------------- => ground truth: 9999mmmmrrrriiiiffffnnnn6666eeee
network out: -------------------------------- => ground truth: ppppqqqq11113333vvvv999922228888
network out: -------------------------------- => ground truth: 5555oooo4444yyyy22228888eeee7777
network out: -------------------------------- => ground truth: aaaa----bbbb3333zzzzooooccccqqqq
network out: -------------------------------- => ground truth: 9999mmmm666688885555zzzz3333iiii
network out: -------------------------------- => ground truth: 8888rrrraaaannnnwwwwffffssssffff
network out: -------------------------------- => ground truth: zzzzjjjj22227777ttttttttrrrrbbbb
network out: -------------------------------- => ground truth: tttt8888ssssssssppppddddrrrrgggg
network out: -------------------------------- => ground truth: bbbb9999aaaawwwweeeeggggeeeellll
network out: -------------------------------- => ground truth: kkkkyyyyhhhhuuuuppppjjjjtttt1111
network out: -------------------------------- => ground truth: ----hhhh1111hhhhsssswwwwhhhhtttt
network out: -------------------------------- => ground truth: 2222aaaa44449999rrrrgggg22224444
network out: -------------------------------- => ground truth: kkkkxxxxhhhhyyyyiiiieeeessssaaaa
 Minibatch[1101-1150]: loss = 2.656250 * 102400, metric = 100.00% * 102400;
network out: -------------------------------- => ground truth: 22229999nnnnkkkkggggffff----nnnn
network out: -------------------------------- => ground truth: ggggwwwwrrrr6666pppp22225555wwww
network out: -------------------------------- => ground truth: pppp2222ttttkkkkjjjj3333oooo0000
network out: -------------------------------- => ground truth: ----cccckkkkssssssss555577774444
network out: -------------------------------- => ground truth: aaaaggggaaaawwww5555jjjjvvvv0000
network out: -------------------------------- => ground truth: ddddoooo999944446666nnnnllll7777
network out: -------------------------------- => ground truth: zzzzdddd33336666jjjjiiiijjjjdddd
network out: -------------------------------- => ground truth: 3333ffff3333yyyyxxxxssssaaaa1111
network out: -------------------------------- => ground truth: zzzzxxxxllllyyyydddd1111rrrryyyy
network out: -------------------------------- => ground truth: 6666cccccccc8888lllloooo2222jjjj
network out: -------------------------------- => ground truth: 2222wwwwjjjj3333----22223333----
network out: -------------------------------- => ground truth: wwww6666xxxx----vvvvnnnnuuuuiiii
network out: -------------------------------- => ground truth: 1111nnnn8888pppprrrrhhhh3333uuuu
network out: -------------------------------- => ground truth: jjjj6666yyyytttt44441111pppp3333
network out: -------------------------------- => ground truth: wwwwnnnneeee----oooouuuu3333vvvv
```

**Comment by sayanpa** [comment]
Have you tried with a smaller learning rate. Try the different learner options. 

https://github.com/Microsoft/CNTK/blob/master/Manual/Manual_How_to_use_learners.ipynb

**Comment by cha-zhang** [comment]
Close this issue due to lack of response. Please feel free to reopen.
